### ISSUE TITLE
KnativeEventing deployment fails when GithubSource is enabled

### ISSUE THREAD

**Issue by sameersbn** [knative/operator#1079]
**Describe the bug**

Knative Eventing deployment fails if the GithubSource is enabled in the operator.

**Expected behavior**

GitHubSource controller should be deployed and Knative Eventing should deploy without any errors

**To Reproduce**

Deploy KnativeEventing on a k8s 1.22 cluster using the following:

```yaml
apiVersion: operator.knative.dev/v1beta1
kind: KnativeEventing
metadata:
  name: knative-eventing
  namespace: knative-eventing
spec:
  # ...
  source:
    github:
      enabled: true
```

**Knative release version**

1.2.0 and above

**Additional context**

FWIW, I am using a EKS 1.22 cluster. However I have observed the same issue on a minikube cluster.

```console
$ kubectl -n knative-eventing describe KnativeEventing knative-eventing
...
Status:
  Conditions:
    Last Transition Time:  2022-05-18T09:12:24Z
    Status:                True
    Type:                  DependenciesInstalled
    Last Transition Time:  2022-05-18T09:12:25Z
    Status:                True
    Type:                  DeploymentsAvailable
    Last Transition Time:  2022-05-19T05:29:51Z
    Message:               Install failed with message: no matches for kind "CustomResourceDefinition" in version "apiextensions.k8s.io/v1beta1"
    Reason:                Error
    Status:                False
    Type:                  InstallSucceeded
    Last Transition Time:  2022-05-19T05:29:51Z
    Message:               Install failed with message: no matches for kind "CustomResourceDefinition" in version "apiextensions.k8s.io/v1beta1"
    Reason:                Error
    Status:                False
    Type:                  Ready
    Last Transition Time:  2022-05-18T09:11:49Z
    Status:                True
    Type:                  VersionMigrationEligible
  Manifests:
    /var/run/ko/knative-eventing/1.2.2
  Observed Generation:  5
  Version:              1.2.2
Events:
  Type     Reason         Age                 From                        Message
  ----     ------         ----                ----                        -------
  Warning  InternalError  15s (x46 over 14h)  knativeeventing-controller  failed to apply non rbac manifest: no matches for kind "CustomResourceDefinition" in version "apiextensions.k8s.io/v1beta1"
```

**Comment by houshengbo** [comment]
apiextensions.k8s.io/v1beata1 is deprecated in k8s v1.22.0 version.
Hmm, I need to talk with the eventing source team about it.

**Comment by lionelvillard** [comment]
fixed in the main branch: https://github.com/knative-sandbox/eventing-github/pull/229

Here is the backport PR to 1.4: https://github.com/knative-sandbox/eventing-github/pull/287. I just need to figure out how to get through the easyCLA guard.

**Comment by houshengbo** [comment]
@sameersbn operator 1.5 has released, I am sure it is resolved in this version.

There will be a patch release for 1.4 as well in 24 hours I guess. It will be resolved in 1.4.2 as well yet to come.
